# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: toolkit/components/perfmonitoring/PerformanceUtils.cpp
# Commit: 47f5698d6c72
# Full Hash: 47f5698d6c72d8033841e708b6df53732393b5b3
# Author: Nika Layzell <nika@thelayzells.com>
# Date: 2020-04-24 09:51:27
# Regressor Bug: 1580565
# File Overlap Count: 2
# Description:
#   Bug 1580565 - Part 6: Add a unique ID to each BrowsingContextGroup, r=kmag
#   
#   This allows us to explicitly specify BrowsingContextGroups when synchronizing
#   them. A major advantage of this is that it means we can handle an attempt to
#   create a BrowsingContext with a parent which the content process is unaware of,
# ==============================================================================

diff -r e1802670dcc4 -r 47f5698d6c72 toolkit/components/perfmonitoring/PerformanceUtils.cpp
--- a/toolkit/components/perfmonitoring/PerformanceUtils.cpp	Thu Apr 23 21:52:49 2020 +0000
+++ b/toolkit/components/perfmonitoring/PerformanceUtils.cpp	Thu Apr 23 21:52:51 2020 +0000
@@ -41,12 +41,7 @@
   }
 
   nsTArray<RefPtr<BrowsingContextGroup>> groups;
-  if (XRE_IsContentProcess()) {
-    groups.AppendElements(
-        ContentChild::GetSingleton()->BrowsingContextGroups());
-  } else {
-    groups.AppendElements(ContentParent::BrowsingContextGroups());
-  }
+  BrowsingContextGroup::GetAllGroups(groups);
 
   nsTArray<DocGroup*> docGroups;
   for (auto& browsingContextGroup : groups) {